### DISCUSSION TITLE
Feature request: much easier module level searching when you need to search multiple columns

### DISCUSSION THREAD

**Discussion by adam-jones-net** [area17/twill#1124] [Ideas]
I've been really surprised how difficult it is to make module level searches work (ie using the search box when you are in a model index page).   The issue is that as soon as you need the search to run across multiple columns you hit a problem. 

It seems simple according to the documentation, ie you setup $defaultFilters in your controller and you're away.  Only its not that simple at all because the resulting sql queries generated by defining multiple fields in $defaultFilters are all using the AND operator, which for me is a very odd choice as once you are using multiple fields, the probability that the search query is going to be across all of the columns is very low.  

So all I'm really requesting is that some extra option be added to allow the ANDs to become ORs.    I now have a solution for doing this manually in code which works fine [see here](https://github.com/area17/twill/issues/360), but it seems like for something as fundamental as search that this shouldn't be required.

The global search in twill defined via the 'search_fields' attribute in the dashboard/modules part of the twill config file works with OR operators out of the box and so it seems inconsistent that it's so much more work to get this other search system functioning.

**Comment by ifox** [comment]
I'm not sure why you are not using the `searchIn` helper I linked an example to above in that thread. It does exactly what you need in a single line, and does indeed use a OR, not a AND.

**Comment by adam-jones-net** [comment]
Ah yes I had seen that a while back.  I'd still argue this is something that should be considered a feature request.

It's definitely a few less lines of code to do it that way agreed easier @ifox 

Still, considering other things to do with basic search setup are possible via a setting in a variable it seems like this is somewhat of a peculiarity that the model level search works with AND as a default operator and that you need to code up something to change that.  

I'd argue that nearly everyone would assume that OR would be used as an operator as standard.  Right now, looking in the documentation its somewhat deceptive that it appears simple to setup.

I.e. you define $defaultFilters and thats it.  However that isn't the case and it seems that there is no really clear documentation stating that it works with AND as default, this is how you change it etc etc.   

I'm just arguing it as a feature request for the sake of consistency really.
